### PR TITLE
Fix issues

### PR DESCRIPTION
**Pull request by sbs2001**
Fix https://github.com/buixor/cs-blocklist-mirror/issues/2 https://github.com/buixor/cs-blocklist-mirror/issues/3 https://github.com/buixor/cs-blocklist-mirror/issues/4 https://github.com/buixor/cs-blocklist-mirror/issues/5 https://github.com/buixor/cs-blocklist-mirror/issues/6

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -22,8 +23,12 @@ var globalDecisionRegistry DecisionRegistry = DecisionRegistry{
 
 func runServer(cfg BouncerConfig) {
 	for _, blockListCFG := range cfg.Blocklists {
-		f := getHandlerForBlockList(blockListCFG)
+		f, err := getHandlerForBlockList(blockListCFG)
+		if err != nil {
+			log.Fatal(err)
+		}
 		http.HandleFunc(blockListCFG.Endpoint, f)
+		prometheus.MustRegister(RouteHits)
```
Comment (buixor) [inline main.go:31]: not sure what happens if we register it in a loop

File: server.go [hunk 2]
```diff
@@ -66,18 +76,65 @@ func satisfiesBasicAuth(r *http.Request, user, password string) bool {
...
-		apiCountersByFormatName[blockListCfg.Format].Inc()
-		if strings.EqualFold(blockListCfg.Authentication.Type, "basic") {
-			if !satisfiesBasicAuth(r, blockListCfg.Authentication.User, blockListCfg.Authentication.Password) {
-				http.Error(w, "access denied", http.StatusForbidden)
-				return
+		RouteHits.WithLabelValues(
+			blockListCfg.Endpoint,
+		).Inc()
+
+		ip, _, _ := net.SplitHostPort(r.RemoteAddr)
```
Comment (buixor) [inline server.go:123]: maybe don't drop err :)

File: server.go [hunk 3]
```diff
@@ -66,18 +76,65 @@ func satisfiesBasicAuth(r *http.Request, user, password string) bool {
...
+		if !networksContainIP(trustedIPs, ip) {
```
Comment (buixor) [inline server.go:124]: to make the logic more trivial, don't we want to have an type of auth like "ip_based" ?

**Review by buixor — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (sbs2001) [reply main.go]: # HELP blocklist_requests_total Number of calls to each blocklist
# TYPE blocklist_requests_total counter
blocklist_requests_total{route="/security/blocklist"} 3
blocklist_requests_total{route="/security/blocklist2"} 4

File: server.go (hunk 2 above)
Comment (sbs2001) [reply server.go]: done

File: server.go (hunk 3 above)
Comment (sbs2001) [reply server.go]: done

**Pull request merged**